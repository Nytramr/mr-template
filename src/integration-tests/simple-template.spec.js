import { Template } from '../index';

describe('Simple templates', () => {
  test('no actions template', async () => {
    const templateTpl =
      'This is a template' +
      '\n' +
      'With no Actions' +
      '\n' +
      'Some "strings"' +
      '\n' +
      '\n' +
      'And some scaped Actions' +
      '\n' +
      '\n' +
      '{\\{ this is not an action }}' +
      '\n';

    const template = new Template('My template');
    expect(() => template.parse(templateTpl)).not.toThrow();

    expect(template.execute()).toMatchSnapshot();
  });

  test('Hello world template', async () => {
    const template = new Template('My template');
    expect(() => template.parse('Hello {{.}}!')).not.toThrow();

    expect(template.execute('world')).toBe('Hello world!');
    expect(template.execute('John')).toBe('Hello John!');
  });

  test('Template with comments', async () => {
    const templateTpl =
      '{{/* This is a comment */}}' +
      '{{/* This is also a comment */ -}}\n' +
      "Hello{{' '}}" +
      '\n' +
      '{{- /* This is another comment */}}' +
      '{{.}}\n' +
      '{{- /* This is yet another comment */ -}}' +
      '\n!';

    const template = new Template('My template');
    expect(() => template.parse(templateTpl)).not.toThrow();

    expect(template.execute('world')).toBe('Hello world!');
    expect(template.execute('John')).toBe('Hello John!');
  });

  test('Text and spaces template', async () => {
    const template = new Template('My template');
    expect(() => template.parse('{{23 -}} < {{- 45}}')).not.toThrow();

    expect(template.execute()).toBe('23<45');
  });
});
