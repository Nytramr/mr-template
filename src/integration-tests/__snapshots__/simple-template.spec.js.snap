// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple templates no actions template 1`] = `
"This is a template
With no Actions
Some "strings"

And some scaped Actions

{\\{ this is not an action }}
"
`;
